haystack/schema.py

```diff
@@ -114,6 +116,9 @@ def __init__(self, question: str,
         else:
             self.id = str(uuid4())
 
+        if not created_at:
```

Comment:
Should we use database-level timestamps instead of setting them here? The application layer can pass a null value and the corresponding database populates it when inserting the document. For the `SQLDocumentStore`, we already have them configured as `server_default`.

Reply:
Hmm... I thought it would be simpler to have it on the application layer as the integration on database-level might vary a lot between document stores. For example, Elasticsearch deprecated the simple "automatic timestamps" (https://stackoverflow.com/questions/56923174/elasticsearch-how-to-add-created-at-and-updated-at-timestamps). Now you can do it via ingest pipelines, but I am not sure how easy that is to configure from the application layer (Haystack). Especially, in some enterprise setups where you just get order an "Elasticsearch service" from your IT department you might only have access to an index but not the permissions to create ingest pipelines...

Reply:
In that case, we could consider doing it at the document store level. In any case, I think it should not make any big difference 🙂

Reply:
Ok makes sense, moved it to the docstore / DB level.